core/src/main/java/io/grpc/internal/PickFirstLeafLoadBalancer.java

```diff
@@ -180,10 +180,18 @@ public Status acceptResolvedAddresses(ResolvedAddresses resolvedAddresses) {
 
   @Override
   public void handleNameResolutionError(Status error) {
+    if (rawConnectivityState == SHUTDOWN) {
```

Comment:
Was this introduced to avoid NPE and handle the case of acceptResolvedAddresses being called when the state is SHUTDOWN ? Can we add test for this as well?

Reply:
Eric pointed out that it shouldn't be possible to get into that state, so I don't think we can add a test.  I was being overly cautious in case someone messed up the logic in the future.